distribution/lib/Standard/Base/0.0.0-dev/src/System/File_Format.enso

```diff
@@ -52,12 +68,31 @@ type Plain_Text_Format
             ".log" -> Plain_Text_Format.Plain_Text
             _ -> Nothing
 
+    ## If the File_Format supports reading from the web response, return a configured instance.
+    for_web : Text -> URI -> Plain_Text_Format | Nothing
+    for_web content_type _ =
+        parts = content_type.split ";" . map .trim
+        case parts.first of
+            "text/plain" ->
+                charset_part = parts.find (x->x.starts_with "charset=")
+                encoding = if charset_part.is_error then Encoding.utf_8 else
+                    parsed = Encoding.from_name (charset_part.drop 8)
+                    if parsed.is_error then Encoding.utf_8 else parsed
```

Comment:
I'm a bit hesitant with falling back to UTF-8 without a blink of an eye.

Ideall, I think I'd error if charset cannot be figured out and ask the user to specify it explicitly. But I guess since UTF-8 is so prevalent, falling back to it _does_ make sense. Still, I think we should at least attach a warning here.

Reply:
Our current approach treats the web as UTF-8. This improved on that by attempting to read the charset. We will attach a warning if the decoding fails.

Reply:
Ok, fair.